ticker_service_test.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func TestTickerService_Start_SendDeviceMeasurement(t *testing.T) {
+	dao := inMemoryDeviceDAO{devices: []Device{{Id: 0, Interval: 1, Value: 5}}}
+	ds := DeviceService{dao: &dao}
+	measurements := make(chan Measurement)
+	defer close(measurements)
+
+	underTest := TickerService{ds: &ds, measurements: measurements}
+	defer underTest.Stop()
+
+	timeout := time.After(3 * time.Second)
+	done := make(chan bool)
+	defer close(done)
+
+	var result Measurement
+
+	go func() {
+		_ = underTest.Start()
+		result = <-measurements
+		done <- true
+	}()
+
+	select {
+	case <-timeout:
```

Comment:
I know it will work and everything will be fine but you are relying on some arbitrarily set timeout. In future cases it might not be that easy to set the right timeout (like for example on loaded jenkins machine it could take longer).

My suggestion for the future - just don't bother with the timeout. I would just set timeout for the whole test run (for example with `go test -timeout` or in the build pipeline) and in the test I would just have `<-done` - simplifies the test and still works, right?